server/app/files.go

```diff
@@ -1,34 +1,91 @@
 package app
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"path/filepath"
 	"strings"
 
+	"github.com/mattermost/mattermost-server/v6/model"
+
 	"github.com/mattermost/focalboard/server/utils"
 
 	"github.com/mattermost/mattermost-server/v6/shared/mlog"
 
 	"github.com/mattermost/mattermost-server/v6/shared/filestore"
 )
 
+var errEmptyFilename = errors.New("IsFileArchived: empty filename not allowed")
+
 func (a *App) SaveFile(reader io.Reader, workspaceID, rootID, filename string) (string, error) {
 	// NOTE: File extension includes the dot
 	fileExtension := strings.ToLower(filepath.Ext(filename))
 	if fileExtension == ".jpeg" {
 		fileExtension = ".jpg"
 	}
 
-	createdFilename := fmt.Sprintf(`%s%s`, utils.NewID(utils.IDTypeNone), fileExtension)
-	filePath := filepath.Join(workspaceID, rootID, createdFilename)
+	createdFilename := utils.NewID(utils.IDTypeNone)
+	fullFilename := fmt.Sprintf(`%s%s`, createdFilename, fileExtension)
+	filePath := filepath.Join(workspaceID, rootID, fullFilename)
 
-	_, appErr := a.filesBackend.WriteFile(reader, filePath)
+	fileSize, appErr := a.filesBackend.WriteFile(reader, filePath)
 	if appErr != nil {
 		return "", fmt.Errorf("unable to store the file in the files storage: %w", appErr)
 	}
 
-	return createdFilename, nil
+	now := utils.GetMillis()
+
+	fileInfo := &model.FileInfo{
+		Id:              createdFilename[1:],
+		CreatorId:       "boards",
+		PostId:          " ",
+		ChannelId:       " ",
+		CreateAt:        now,
+		UpdateAt:        now,
+		DeleteAt:        0,
+		Path:            " ",
+		ThumbnailPath:   " ",
+		PreviewPath:     " ",
+		Name:            filename,
+		Extension:       fileExtension,
+		Size:            fileSize,
+		MimeType:        " ",
+		Width:           0,
+		Height:          0,
+		HasPreviewImage: false,
+		MiniPreview:     nil,
+		Content:         "",
+		RemoteId:        nil,
+	}
```

Comment:
Saving a string with a space (" "), seems suspicious to me. I assume an empty string ("") won't work. I would almost prefer "empty" or "unused" to a space string.

Reply:
Yeah, an empty string causes Mattermost server to crash when retrieving fileinfo from DM. The columns are nullable in DB but not nullable in go structs, hence the dummy string.

I've replaced this with `"empty"`.